bip-taro.mediawiki

```diff
@@ -470,6 +492,23 @@ other normal script-path scripts within the tapscript tree.
 The above example creates only a single asset. It's possible to also create
 several assets within a single output, thereby batching asset creation.
 
+===Asset Burning===
+
+Given the requirement that Taro commitments be unique within the context of a
+tapscript commitment, Taro assets can also be ''provably burned''.
+
+In order to burn an asset, the holder of the asset can simply ''remove'' the
+Taro asset leaf from their asset tree.
+
+In order to prove to a 3rd party that an asset has been burnt, then holder of
+said asset must reveal the following information:
+* A non-inclusion proof rooted at their asset tree that proves that asset in question is no longer committed to.
+* A proof for each other top-level output in the burning transaction that assets the following:
+** For each output, the first depth of the control block leaf reveal must be presented:
+*** If the output key was created via BIP 86, then simply verify the mapping based on the internal key. In this case there is no script tree being committed to.
+*** If the proof is of depth 0, meaning only a single value is committed to, then assert the item isn't a valid Taro asset root.
+*** If the proof is of depth 1, then assert that either both elements are tap branches, or one of the items is a leaf that isn't a valid Taro asset root.
```

Comment:
What if we want to burn some assets and not all? You would still want a new Taro commitment to exist excluding the burnt assets.

Reply:
I think that's covered under the normal interactive transfer flow: you'd show the normal proofs of non-inclusion within the tapscript tree. This here is to ensure that things aren't committed elsewhere, which also needs to be propagated into the internal transfer case.